src/modules/date/index.ts

```diff
@@ -387,14 +391,15 @@ export class SimpleDateModule {
         },
     legacyTo?: string | Date | number
   ): Date {
-    if (typeof options !== 'object' || options instanceof Date) {
+    if (options instanceof Date || typeof options !== 'object') {
       deprecated({
         deprecated: 'faker.date.between(from, to)',
         proposed: 'faker.date.between({ from, to })',
         since: '8.0',
         until: '9.0',
       });
-      options = { from: options, to: legacyTo };
+      // We use options as fallback for legacyTo avoid TS errors for unintended usage.
+      options = { from: options, to: legacyTo ?? options };
```

Comment:
Can we extend the comment to remove this workaround in v9?

```suggestion
      // We use options as fallback for legacyTo avoid TS errors for unintended usage.
      // TODO: Remove in v9
      options = { from: options, to: legacyTo ?? options };
```

Reply:
The whole deprecation block will be gone in v9.